plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/RawSystemTableProvider.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.base.VerifyException;
+import com.google.inject.Inject;
+import io.trino.spi.connector.ConnectorMetadata;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.connector.SystemTableProvider;
+import io.trino.spi.connector.TableNotFoundException;
+import io.trino.spi.type.TypeManager;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+
+import java.util.Optional;
+
+import static io.trino.plugin.iceberg.IcebergTableName.isDataTable;
+import static io.trino.plugin.iceberg.IcebergTableName.isIcebergTableName;
+import static io.trino.plugin.iceberg.IcebergTableName.isMaterializedViewStorage;
+import static io.trino.plugin.iceberg.IcebergTableName.tableNameFrom;
+import static java.util.Objects.requireNonNull;
+
+public class RawSystemTableProvider
+        implements SystemTableProvider
+{
+    private final TypeManager typeManager;
+
+    @Inject
+    public RawSystemTableProvider(TypeManager typeManager)
+    {
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+    }
+
+    @Override
+    public Optional<SchemaTableName> getSourceTableName(SchemaTableName table)
+    {
+        try {
+            String name = tableNameFrom(table.getTableName());
+            return Optional.of(new SchemaTableName(table.getSchemaName(), name));
+        }
+        catch (IllegalArgumentException e) {
+            return Optional.empty();
+        }
+    }
```

Comment:
I'm not exactly sure how that interface works - once we derive the base name, is `getSystemTable` called with the *original* or *derived* table name?

Reply:
`getSystemTable` is called with original `SchemaTableName`. `getSourceTableName` is used for access checks (at least in hive).